tests/components/shopping_list/test_config_flow.py

```diff
@@ -0,0 +1,38 @@
+"""Test config flow."""
+
+from homeassistant import data_entry_flow
+from homeassistant.components.shopping_list.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.setup import async_setup_component
+
+
+async def test_manual_config_set(hass):
+    """Test entry will be imported."""
+    assert await async_setup_component(hass, DOMAIN, {})
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, data={}, context={"source": SOURCE_USER}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+
+
+async def test_user(hass):
+    """Test we can start a config flow."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, data=None, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_user_confirm(hass):
+    """Test we can finish a config flow."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, data={}, context={"source": SOURCE_USER}
```

Comment:
With the previous comment, this will be easier to read : 
```suggestion
        DOMAIN, context={"source": SOURCE_USER}, data={}
```